src/commands/service/delete.ts

```diff
@@ -0,0 +1,59 @@
+import { Command, CliUx } from '@oclif/core'
+import { Jamsocket } from '../../jamsocket'
+import * as inquirer from 'inquirer'
+import { formatDistanceToNow } from 'date-fns'
+
+export default class Create extends Command {
+  static description = 'Deletes a service'
+
+  static examples = [
+    '<%= config.bin %> <%= command.id %> my-service',
+  ]
+
+  static args = [{ name: 'name', required: true }]
+
+  public async run(): Promise<void> {
+    const { args } = await this.parse(Create)
+
+    const jamsocket = Jamsocket.fromEnvironment()
+    const serviceInfo = await jamsocket.serviceInfo(args.name)
+
+    const spawnTokenCount = serviceInfo.spawn_tokens_count
+    const lastImgUpload = serviceInfo.last_image_upload_time ? formatDistanceToNow(new Date(serviceInfo.last_image_upload_time)) : null
+    const lastSpawn = serviceInfo.last_spawned_at ? formatDistanceToNow(new Date(serviceInfo.last_spawned_at)) : null
+
+    const formattedInfo: string[] = [
+      `The service ${serviceInfo.name}`,
+      lastImgUpload === null ? 'has no images,' : `had an image pushed to it ${lastImgUpload} ago,`,
+      spawnTokenCount === 0 ? 'has no spawn tokens,' : `has ${spawnTokenCount} spawn tokens,`,
+      lastSpawn === null ? 'and has never been spawned.' : `and was last spawned ${lastSpawn} ago.`,
+    ]
+
+    this.log()
+    this.log(formattedInfo.join(' '))
+    this.log(`(For more information about this service, run \`jamsocket service info ${serviceInfo.name}\`.)\n`)
+
+    const response = await inquirer.prompt([{
+      name: 'confirmDelete',
+      message: 'Are you sure you want to delete this service? This action cannot be undone!',
+      type: 'list',
+      choices: [{ name: 'no' }, { name: 'yes' }],
+    }])
+
+    if (response.confirmDelete !== 'yes') {
+      this.log('Service deletion canceled.')
+      return
+    }
+
+    this.log()
+    const userInput = await CliUx.ux.prompt(`Type the service name to delete it (${serviceInfo.name})`)
```

Comment:
Maybe we skip the "type service name" part if `lastImgUpload` is null?

Reply:
I thought about this before but was kinda torn about it because it could make the CLI behave unpredictably to the user. "Why does it sometimes prompt me and other times it doesn't?" I think I prefer to have commands behave consistently / predictably. One way we could do it, though, is if we add some context to the prompt, the either hints or outright explains that we're prompting because there are resources attached to the given service? 🤔

Reply:
What about making it either prompt for a yes/no or for the full name, depending on it? And making it clear that entering anything else will abort?

I'm fine with it as-is, though.

Reply:
Thought through it a bit more, and I think it's just simpler to leave it as is for now.